elements/lisk-p2p/test/integration/unseeded_network.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import { expect } from 'chai';
+import { P2P } from '../../src/index';
+import { wait } from '../utils/helpers';
+import { platform } from 'os';
+
+describe('Unseeded network: Each node has an empty seedPeers list', () => {
+	let p2pNodeList: ReadonlyArray<P2P> = [];
+	const NETWORK_START_PORT = 5000;
+	const NETWORK_PEER_COUNT = 10;
+	const POPULATOR_INTERVAL = 50;
+	const DEFAULT_MAX_OUTBOUND_CONNECTIONS = 20;
+	const DEFAULT_MAX_INBOUND_CONNECTIONS = 100;
+	const ALL_NODE_PORTS: ReadonlyArray<number> = [
+		...new Array(NETWORK_PEER_COUNT).keys(),
+	].map(index => NETWORK_START_PORT + index);
+	const NO_PEERS_FOUND_ERROR = `Request failed due to no peers found in peer selection`;
+
+	before(async () => {
+		sandbox.restore();
+	});
+
+	beforeEach(async () => {
+		// Make sure that integration tests use real timers.
+		sandbox.restore();
+		p2pNodeList = ALL_NODE_PORTS.map(nodePort => {
+			return new P2P({
+				connectTimeout: 200,
+				wsEngine: 'ws',
+				populatorInterval: POPULATOR_INTERVAL,
+				maxOutboundConnections: DEFAULT_MAX_OUTBOUND_CONNECTIONS,
+				maxInboundConnections: DEFAULT_MAX_INBOUND_CONNECTIONS,
+				nodeInfo: {
+					wsPort: nodePort,
+					nethash:
+						'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
+					version: '1.0.1',
+					protocolVersion: '1.1',
+					minVersion: '1.0.0',
+					os: platform(),
+					height: 0,
+					broadhash:
+						'2768b267ae621a9ed3b3034e2e8a1bed40895c621bbb1bbd613d92b9d24e54b5',
+					nonce: `O2wTkjqplHII${nodePort}`,
+				},
+			});
+		});
+		await Promise.all(p2pNodeList.map(async p2p => await p2p.start()));
+		await wait(100);
+	});
+
+	afterEach(async () => {
+		await Promise.all(
+			p2pNodeList
+				.filter(p2p => p2p.isActive)
+				.map(async p2p => await p2p.stop()),
+		);
+		await wait(1000);
+	});
+
+	it('should set the isActive property to true for all nodes', () => {
+		for (let p2p of p2pNodeList) {
+			expect(p2p).to.have.property('isActive', true);
+		}
+	});
+
+	describe('P2P.request', () => {
```

Comment:
Do we need this describe here? I think it doesn't give us so much value. Perhaps we can remove it and fine a better description for the inside test.